pkg/hub/client/public/filter_test.go

```diff
@@ -159,9 +160,127 @@ func TestImplementationRequirementsSatisfiedByFilters(t *testing.T) {
 			},
 			requirementsSatisfiedBy: []*gqlpublicapi.TypeInstanceValue{
 				{
-					TypeRef: &gqlpublicapi.TypeReferenceWithOptionalRevision{
+					TypeRef: &gqlpublicapi.TypeReferenceInput{
 						Path:     "cap.type.gcp.sa",
-						Revision: ptr.String("0.1.1"),
+						Revision: "0.1.1",
+					},
+				},
+			},
+		},
+		{
+			name: "Return Implementations satisfied by injected RequiredTypeInstances",
+			expRevision: []gqlpublicapi.ImplementationRevision{
+				fixImplementationRevision("without-any-requirements", "0.0.1"),
+				fixImplementationRevisionWithRequire("with-gcp-sa-requirement", "0.1.0", gqlpublicapi.ImplementationRequirement{
+					Prefix: "cap.core.type.platform",
```

Comment:
This prefix doesn't fit here. I checked and it's unused. Maybe it will be better to get rid of it because it was really confusing for me as based on the API I tried to create such fixture:
```go
				Prefix: "cap.core.type.platform",
					AllOf: []*gqlpublicapi.ImplementationRequirementItem{
						{
							TypeRef: &gqlpublicapi.TypeReference{
								Path:     "kubernetes",
								Revision: "0.1.1",
							},
						},
					},
```